manifests/concourse-manifest/extensions/datadog.yml

```diff
@@ -0,0 +1,10 @@
+jobs:
+  - name: concourse
+    templates:
+      - name: datadog-agent
```

Comment:
In this case we don't care about specific order, so we could use (( merge )). As it turns out

> If you don't specify a specific merge strategy, the array will be merged automatically; using keys if they exist (i.e. (( merge )), and array indices otherwise ((( inline ))).

This has been tested to work correctly without (( append )), so I don't see the need to include it here.